src/types.ts

```diff
@@ -29,8 +65,8 @@ export type AnyParamConstructor<T> = new (...args: any) => T;
 /**
  * The Type of a Model that gets returned by "getModelForClass" and "setModelForClass"
  */
-export type ReturnModelType<U extends AnyParamConstructor<T>, T = any> = ModelType<InstanceType<U>> & U;
-
+export type ReturnModelType<U extends AnyParamConstructor<T>, T = any> = ModelType<InstanceType<U>> | U;
```

Comment:
If this is left as it was before, then no-argument constructors work, but the autocomplete is less clean. If you want to leave no-argument construction support, revert to intersection rather than union.